x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/preview_rules_route.ts

```diff
@@ -173,26 +187,62 @@ export const previewRulesRoute = async (
           }
         };
 
-        const signalRuleAlertType = signalRulesAlertType({
-          ...ruleOptions,
-          lists: context.lists,
-          config,
-          indexNameOverride: previewIndex,
-          ruleExecutionLogClientOverride: previewRuleExecutionLogClient,
-          // unused as we override the ruleExecutionLogClient
-          eventLogService: {} as unknown as IEventLogService,
-          eventsTelemetry: undefined,
-          ml: undefined,
-          refreshOverride: 'wait_for',
-        });
-
-        await runExecutors(
-          signalRuleAlertType.executor,
-          signalRuleAlertType.id,
-          signalRuleAlertType.name,
-          previewRuleParams,
-          alertInstanceFactoryStub
-        );
+        switch (previewRuleParams.type) {
+          case 'query':
+            const queryAlertType = previewRuleTypeWrapper(createQueryAlertType(ruleOptions));
+            await runExecutors(
+              queryAlertType.executor,
+              queryAlertType.id,
+              queryAlertType.name,
+              previewRuleParams,
+              alertInstanceFactoryStub
+            );
+            break;
+          case 'threshold':
+            const thresholdAlertType = previewRuleTypeWrapper(
+              createThresholdAlertType(ruleOptions)
+            );
+            await runExecutors(
+              thresholdAlertType.executor,
+              thresholdAlertType.id,
+              thresholdAlertType.name,
+              previewRuleParams,
+              alertInstanceFactoryStub
+            );
+            break;
+          case 'threat_match':
```

Comment:
I've added the catch for rule type at the top of this file, I think since we intend on having it at some point we could keep this logic here. That way when the switch gets flipped all we need to do is remove the type catch and we're ready to go

Reply:
thanks for this call, that's what I just did <3